src/frontend/src/session.rs

```diff
@@ -471,6 +483,31 @@ impl SessionImpl {
         let db_id = catalog_reader.get_database_by_name(db_name)?.id();
         Ok((db_id, schema.id()))
     }
+
+    pub fn clear_cancel_query_flag(&self) {
+        let mut flag = self.current_query_cancel_flag.write();
+        *flag = None;
+    }
+
+    pub fn reset_cancel_query_flag(&self) -> Tripwire<std::result::Result<DataChunk, BoxedError>> {
+        let mut flag = self.current_query_cancel_flag.write();
+        let (trigger, tripwire) = stream_tripwire(|| Err(Box::new(QueryCancelError) as BoxedError));
+        *flag = Some(trigger);
+        tripwire
+    }
+
+    pub fn cancel_current_query(&self) {
```

Comment:
Is it will cancel all local query in one session?

Reply:
No, only current running query